quartz/components/scripts/search.inline.ts

```diff
@@ -405,11 +406,32 @@ document.addEventListener("nav", async (e: CustomEventMap["nav"]) => {
 
     let searchResults: FlexSearch.SimpleDocumentSearchResultSetUnit[]
     if (searchType === "tags") {
-      searchResults = await index.searchAsync({
-        query: currentSearchTerm.substring(1),
-        limit: numSearchResults,
-        index: ["tags"],
-      })
+      let [tag, ...queries] = currentSearchTerm.substring(1).split(" ")
+      const query = queries.join(" ").trim()
+
+      if (query !== "") {
+        // search by title and content index and then filter by tag (implemented in flexsearch)
+        searchResults = await index.searchAsync({
+          query: query,
+          // search at least 100 documents, so it is enough to filter them by tag (not tags index)
+          limit: Math.max(numSearchResults, 100),
```

Comment:
limit is results, not how many documents to look at so should be safe to keep this at `numSearchResults`

Reply:
yes, that's right, it's the number of results that will be returned
but as far as I understood during testing and from the document search source code(https://github.com/nextapps-de/flexsearch/blob/4c3966709f855c6980c21035cf230b2f9150d065/src/document.js#L592-L635), first the index is searched (taking into account this limit) and only then filtered by tag
as a result, if the index search returns n=limit the most relevant documents, but without the required tag, we will get nothing after filtering

perhaps this is due to the fact that if you first filter by tag and then search by index, you will have to build a new index from all documents with that tag (which is probably too expensive to do in runtime)
but this is just my guess)

Reply:
ah ok, in that case i think we can probably even just remove the limit and then rely on slice on `numSearchResult` in L423 to do that then

Reply:
I did some more testing and research and found a default limit of 100 in the source code of flexsearch https://github.com/nextapps-de/flexsearch/blob/4c3966709f855c6980c21035cf230b2f9150d065/src/document.js#L517 
so if we want to reduce the probability of discarding the right documents with a tag, we need to pass quite a large number (but within reason)
what do you think if we set the limit to 10000?

Reply:
@jackyzha0 can we merge this pr?